static/js/message_list.js

```diff
@@ -254,9 +254,10 @@ exports.MessageList.prototype = {
 
                 // For invite only streams or streams that no longer
                 // exist, hide the resubscribe button
+                // Hide button for guest users
                 var sub = stream_data.get_sub(stream_name);
                 if (sub !== undefined) {
-                    show_button = !sub.invite_only;
+                    show_button = !page_params.is_guest && !sub.invite_only;
```

Comment:
Is `page_params.is_guest` set?  I think we might need to add it.

Reply:
yes, it is set from `fetch_initial_state_data`